src/Macros/BlueprintMacros.php

```diff
@@ -24,5 +24,25 @@ public static function boot(): void
             /** @var class-string<Model> $model */
             return $this->foreignSnowflake($column ?? (new $model())->getForeignKey());
         });
+
+        Blueprint::macro('morphsSnowflake', function (string $name, ?string $indexName = null): void {
+            $this->snowflake("{$name}_id");
+            $this->text("{$name}_type");
+
+            $this->index(
+                ["{$name}_id", "{$name}_type"],
+                $indexName,
+            );
```

Comment:
- the morph index is always (type, id)---order matters in this case
- would prefer this to be a single line

```suggestion
            $this->index(["{$name}_type", "{$name}_id"], $indexName);
```